### ISSUE TITLE
Sum Aggregation  leave out fractional part 

### ISSUE THREAD

**Issue by chroje** [elastic/elasticsearch#30250]
The field name  "zdl" 
"zdl": {
    "type": "float"
}
but I exec sum aggregation show me a integer leave out fractional part
The command is

GET orcl-npmis_zw_ysdfmx_bak/_search
{
  "size": 0,
  "aggs": {
    "a1": {
      "terms": {
        "field": "jldbh.keyword",
        "order": {
          "a2": "desc"
        }
      },
      "aggs": {
        "a2": {
          "sum": {
            "field": "zdl"
          }
        }
      }
    }
  }
}

**Comment by chroje** [comment]
A new question is that the real result is 24662092.64 but show 24662092.75

**Comment by colings86** [comment]
so the sum is actually showing the fractional part now?

The reason the actual sum might be different to the returned sum is detailed here: https://www.elastic.co/guide/en/elasticsearch/reference/6.2/search-aggregations-bucket-terms-aggregation.html#search-aggregations-bucket-terms-aggregation-approximate-counts

(This talks about the doc counts but the same reasoning extends to sub aggregations.

**Comment by chroje** [comment]
But where I use sum Aggregations is Metrics Aggregations instead of Bucket Aggregations, the result still wrong.

exec:
POST /orcl-npmis_zw_ysdfmx_bak/_search
{
  "query": {
    "term": {
      "jldbh": "0634090002765101"
    }
  },
  "aggs": {
    "sum_": {
      "sum": {
        "field": "zdl"
      }
    }
  },
  "_source": "zdl", 
  "size": 100
}

return: 
  "aggregations": {
    "sum_": {
      "value": 24662092.75
    }
  }

And the real rusult is 24662092.64

Some result have the fractional part but some don`t .
In other words,some result leave out the fractional part.

By the way,my data is from oracle by logstash , does it matter?

**Comment by colings86** [comment]
> But where I use sum Aggregations is Metrics Aggregations instead of Bucket Aggregations,

Yes but the sum aggregation is under the terms agg so as I explained before the fact that the terms aggregation is approximate (due to the reasons in the linked documentation) means that the sum aggregation (and any aggregations under the terms aggregation) will be approximate too. Note that you can use the `doc_count_error_upper_bound` in the terms aggregation response to determine how many documents might be missed and the `shard_size` parameter on the terms aggregation to reduces the error by retrieving more buckets from the shards (also explained in the same documentation page).

> Some result have the fractional part but some don`t .

This is probably due to the way JSON is encoded. The JSON spec only has a number type which is a floating point number it does not have an explicit integer type. A lot of JSON encoders will only print a fractional part if there is a fractional part, so I think these values you are seeing without a fractional part are just where the sum adds up to a whole number.

Since there doesn't seem to be a bug here I am going to close this issue, we can always reopen it if it turns out there is a bug here later. 

If you have more questions about using Elasticsearch please ask them on [the forum](https://discuss.elastic.co/new) which is more set up for answering questions, we like to keep this issues list for bug reports and feature requests.

**Comment by chroje** [comment]
thx
